flow/TLSConfig.actor.cpp

```diff
@@ -823,27 +1000,23 @@ bool TLSPolicy::verify_peer(bool preverified, X509_STORE_CTX* store_ctx) {
 		return false;
 	}
 
-	if (!rules.size()) {
-		return true;
-	}
+	PeerVerifier verifier(rules, store_ctx, is_client);
 
-	// Any matching rule is sufficient.
-	for (auto& verify_rule : rules) {
-		std::tie(verify_success, verify_failure_reason) = check_verify(&verify_rule, store_ctx, is_client);
-		if (verify_success) {
-			rc = true;
-			break;
-		} else {
-			if (verify_failure_reason.length() > 0)
-				verify_failure_reasons.insert(verify_failure_reason);
-		}
-	}
+	if (verifier.isErr()) {
+		// Must have all rules tried with failure
+		ASSERT_EQ(verifier.getFailureReasons().size(), rules.size());
 
-	if (!rc) {
-		// log the various failure reasons
-		for (std::string reason : verify_failure_reasons) {
-			TraceEvent(SevWarn, "TLSPolicyFailure").suppressFor(1.0).detail("Reason", reason);
+		for (size_t i = 0; i < rules.size(); ++i) {
+			const auto& failureReason = verifier.getFailureReasons()[i];
+			const auto& rule = rules[i];
+			TraceEvent(SevWarn, "TLSPolicyFailure")
```

Comment:
Because of suppression, I think only the first rule will be printed. The right approach is to create the trace event outside the loop and add each rule as a field. Probably the event buffer size should be expanded.